test/MUXControlsTestApp/InfoBar.cs

```diff
@@ -0,0 +1,473 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI;
+using System.Windows.Input;
+using Windows.Foundation;
+using Windows.UI.Composition;
+
+namespace MUXControlsTestApp
+{
+    public enum InfoBarCloseReason
+    {
+        CloseButton,
+        Programattic
+    }
+    public enum InfoBarSeverity
+    {
+        Critical,
+        Warning,
+        Informational,
+        Success,
+        Default,
+        None
+    }
+
+    public class InfoBarClosedEventArgs : EventArgs
+    {
+        public InfoBarCloseReason Reason
+        {
+            get; set;
+        }
+    }
+
+    public class InfoBarClosingEventArgs : EventArgs
+    {
+        public InfoBarCloseReason Reason
+        {
+            get; set;
+        }
+        public bool Cancel
+        {
+            get; set;
+        }
+
+    }
+    public class InfoBarEventArgs : EventArgs
+    {
+        public bool IsHandled
+        {
+            get; set;
+        }
+    }
+    public sealed class InfoBar : ContentControl
+    {
+        Button _actionButton;
+        Button _alternateCloseButton;
+        Button _closeButton;
+        Border _myBorder;
+
+        public event EventHandler<RoutedEventArgs> ActionButtonClick;
+        public event EventHandler<InfoBarEventArgs> CloseButtonClick;
+        public event TypedEventHandler<InfoBar, InfoBarClosedEventArgs> Closed;
+        public event TypedEventHandler<InfoBar, InfoBarClosingEventArgs> Closing;
+
+        private InfoBarCloseReason lastCloseReason = InfoBarCloseReason.Programattic;
+
+
+        public InfoBar()
+        {
+            this.DefaultStyleKey = typeof(InfoBar);
+        }
+
+
+        T GetTemplateChild<T>(string name) where T : DependencyObject
+        {
+            var child = GetTemplateChild(name) as T;
+            if (child == null)
+            {
+                throw new NullReferenceException(name);
```

Comment:
>NullReferenceException [](start = 26, length = 22)

In general we support customers retemplating our controls, it is possible that the template parts you expect wont be present. In that case we shouldn't throw, we should continue doing what we can without the part.

Reply:
Yeah, that makes a lot of sense. I removed the check for whether child is empty, along with the exception.